src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7021,12 +7027,19 @@ private void CheckRuntimeSupportForSymbolAccess(SyntaxNode node, BoundExpression
 
             if (!symbol.RequiresInstanceReceiver())
             {
-                if (instanceReceiver == true)
+                if (instanceReceiver == true && !(IsInsideNameof && IsFeatureAvailable(node, MessageID.IDS_FeatureReducedMemberAccessChecksInNameof)))
```

Comment:
> && !(IsInsideNameof && IsFeatureAvailable(node, MessageID.IDS_FeatureReducedMemberAccessChecksInNameof)) [](start = 44, length = 105)

I think it would be simpler if this didn't check the condition here. Leave this condition as it was, and then return the result of `!CheckFeatureAvailability` below.

Reply:
Or perhaps
```cs
if (CheckFeatureAvailability)
{
    return true;
}
```
to preserve semantics.

---
In reply to: [549459553](https://github.com/dotnet/roslyn/pull/48754#discussion_r549459553) [](ancestors = 549459553)